sdk/tables/azure-data-tables/azure/data/tables/_table_batch.py

```diff
@@ -480,8 +480,9 @@ def _batch_merge_entity(
 
     def delete_entity(
         self,
-        partition_key,  # type: str
-        row_key,  # type: str
+        entity,  # type: Union[TableEntity, Mapping[str, Any]]
+        # partition_key,  # type: str
+        # row_key,  # type: str
```

Comment:
Did you want to try the overloads for this?
I think it would look something like:
```python
from typing import overload

@overload
def delete_entity(self, partition_key: str, row_key: str, **kwargs):
    ...

@overload
def delete_entity(self, entity: Union[TableEntity, Mapping[str, Any]], **kwargs):
    ...

def delete_entity(self, *args, **kwargs):
    try:
        partition_key = args[0]['PartitionKey']
        row_key = args[0]['RowKey']
    except TypeError:
        partition_key = args[0]
        row_key = args[1]
```